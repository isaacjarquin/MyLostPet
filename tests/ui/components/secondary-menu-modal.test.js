import "react-native"
import React from "react"
import SecondaryMenyModal from "../../../components/secondary-menu-modal"
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import renderer from "react-test-renderer"

describe("SecondaryMenyModal", () => {
    const props = {}

    beforeEach(() => {
        wrapper = shallow(<SecondaryMenyModal {...props} />);
    })

    it("renders correctly", () => {
        expect(wrapper).toBeDefined()
        expect(toJSON(wrapper)).toMatchSnapshot();
    })

    describe("_showModal", () => {
        it("shows modal", () => {
            wrapper.setState({ isModalVisible: false })
            wrapper.instance()._showModal()
            expect(wrapper.state().isModalVisible).toBe(true);
        })
    })

    describe("_hideModal", () => {
        it("hides modal", () => {
            wrapper.setState({ isModalVisible: true })
            wrapper.instance()._hideModal()
            expect(wrapper.state().isModalVisible).toBe(false);
        })
    })

    describe("_showPetTypeModal", () => {
        it("shows modal", () => {
            wrapper.setState({ isPetTypeModalVisible: false })
            wrapper.instance()._showPetTypeModal()
            expect(wrapper.state().isPetTypeModalVisible).toBe(true);
        })
    })

    describe("_hidePetTypeModal", () => {
        it("hides modal", () => {
            wrapper.setState({ isPetTypeModalVisible: true })
            wrapper.instance()._hidePetTypeModal()
            expect(wrapper.state().isPetTypeModalVisible).toBe(false);
        })
    })

    describe("_showAutonomousComunityModal", () => {
        it("shows modal", () => {
            wrapper.setState({ isAutonomousComunityModalVisible: false })
            wrapper.instance()._showAutonomousComunityModal()
            expect(wrapper.state().isAutonomousComunityModalVisible).toBe(true);
        })
    })

    describe("_hideAutonomousComunityModal", () => {
        it("hides modal", () => {
            wrapper.setState({ isAutonomousComunityModalVisible: true })
            wrapper.instance()._hideAutonomousComunityModal()
            expect(wrapper.state().isAutonomousComunityModalVisible).toBe(false);
        })
    })

    describe("_showProvinceModal", () => {
        it("shows modal", () => {
            wrapper.setState({ isProvinceModalVisible: false })
            wrapper.instance()._showProvinceModal()
            expect(wrapper.state().isProvinceModalVisible).toBe(true);
        })
    })

    describe("_hideProvinceModal", () => {
        it("hides modal", () => {
            wrapper.setState({ isProvinceModalVisible: true })
            wrapper.instance()._hideProvinceModal()
            expect(wrapper.state().isProvinceModalVisible).toBe(false)
        })
    })

    describe("setType", () => {
        it("sets a type", () => {
            wrapper.setState({ type: "" })
            wrapper.instance().setType("dog")
            expect(wrapper.state().type).toBe("dog")
        })
    })

    describe("setProvince", () => {
        it("sets the province", () => {
            wrapper.setState({ province: "" })
            wrapper.instance().setProvince("Tenerife")
            expect(wrapper.state().province).toBe("Tenerife")
        })
    })

    describe("setAutonomousComunity", () => {
        it("sets autonomous comunity and provincias", () => {
            wrapper.setState({ autonomousComunity: "", provincias: [] })
            wrapper.instance().setAutonomousComunity("Canarias")
            expect(wrapper.state().autonomousComunity).toBe("Canarias")
            expect(wrapper.state().provincias).toEqual([{ "id": 12, "value": "Las Palmas" }, { "id": 13, "value": "Santa Cruz de Tenerife" }])
        })
    })
})
