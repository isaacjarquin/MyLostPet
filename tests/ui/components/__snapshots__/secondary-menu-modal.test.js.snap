// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondaryMenyModal renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "backgroundColor": "#333333",
        "borderRadius": 3,
        "flexDirection": "row",
        "margin": 5,
        "opacity": 0.8,
        "padding": 15,
      }
    }
  >
    <Icon
      color="white"
      name="search"
      raised={false}
      reverse={false}
      reverseColor="white"
      size={30}
      style={
        Object {
          "marginLeft": "20%",
        }
      }
      type="evilIcons"
      underlayColor="white"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignSelf": "center",
          "color": "white",
          "fontSize": 18,
        }
      }
    >
      Buscar mascota
    </Text>
  </TouchableOpacity>
  <ReactNativeModal
    animationIn="slideInUp"
    animationInTiming={300}
    animationOut="slideOutDown"
    animationOutTiming={300}
    avoidKeyboard={false}
    backdropColor="black"
    backdropOpacity={0.7}
    backdropTransitionInTiming={300}
    backdropTransitionOutTiming={300}
    isVisible={false}
    onBackButtonPress={[Function]}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalShow={[Function]}
    style={
      Object {
        "backgroundColor": "#333333",
        "borderTopLeftRadius": 3,
        "borderTopRightRadius": 3,
        "flexDirection": "column",
        "marginTop": 1034,
      }
    }
    useNativeDriver={false}
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderTopLeftRadius": 3,
          "borderTopRightRadius": 3,
          "flexDirection": "row",
          "marginBottom": 10,
          "marginLeft": 15,
          "marginRight": 15,
          "marginTop": 10,
          "opacity": 0.8,
        }
      }
    >
      <Icon
        color="grey"
        name="keyboard-arrow-down"
        raised={false}
        reverse={false}
        reverseColor="white"
        size={30}
        type="MaterialIcons"
        underlayColor="white"
      />
    </TouchableOpacity>
    <View>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "black",
            "borderColor": "#d6d7da",
            "borderWidth": 0.5,
            "flexDirection": "row",
            "opacity": 0.5,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "width": "95%",
            }
          }
        >
          Tipo de mascota
        </Text>
        <Icon
          color="white"
          name="keyboard-arrow-down"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={20}
          type="MaterialIcons"
          underlayColor="white"
        />
      </TouchableOpacity>
      <SecondaryMenuModal
        handler={[Function]}
        hidePetTypeModal={[Function]}
        isVisible={false}
        items={
          Array [
            Object {
              "id": 1,
              "value": "perro",
            },
            Object {
              "id": 2,
              "value": "gato",
            },
            Object {
              "id": 3,
              "value": "conejo",
            },
            Object {
              "id": 4,
              "value": "hamster",
            },
            Object {
              "id": 5,
              "value": "iguana",
            },
            Object {
              "id": 6,
              "value": "hurón",
            },
            Object {
              "id": 7,
              "value": "tortuga",
            },
          ]
        }
      />
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "black",
            "borderColor": "#d6d7da",
            "borderWidth": 0.5,
            "flexDirection": "row",
            "opacity": 0.5,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "width": "95%",
            }
          }
        >
          Comunidad autónoma
        </Text>
        <Icon
          color="white"
          name="keyboard-arrow-down"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={20}
          type="MaterialIcons"
          underlayColor="white"
        />
      </TouchableOpacity>
      <SecondaryMenuModal
        handler={[Function]}
        hidePetTypeModal={[Function]}
        isVisible={false}
        items={
          Array [
            Object {
              "id": "andalucia",
              "provincias": Array [
                Object {
                  "id": 1,
                  "value": "Almería",
                },
                Object {
                  "id": 2,
                  "value": "Cádiz",
                },
                Object {
                  "id": 3,
                  "value": "Córdoba",
                },
                Object {
                  "id": 4,
                  "value": "Granada",
                },
                Object {
                  "id": 5,
                  "value": "Huelva",
                },
                Object {
                  "id": 6,
                  "value": "Jaen",
                },
                Object {
                  "id": 7,
                  "value": "Málaga",
                },
                Object {
                  "id": 8,
                  "value": "Sevilla",
                },
              ],
              "value": "Andalucia",
            },
            Object {
              "id": "aragon",
              "provincias": Array [
                Object {
                  "id": 9,
                  "value": "Huesca",
                },
                Object {
                  "id": 10,
                  "value": "Teruel",
                },
                Object {
                  "id": 11,
                  "value": "Zaragoza",
                },
              ],
              "value": "Aragon",
            },
            Object {
              "id": "canarias",
              "provincias": Array [
                Object {
                  "id": 12,
                  "value": "Las Palmas",
                },
                Object {
                  "id": 13,
                  "value": "Santa Cruz de Tenerife",
                },
              ],
              "value": "Canarias",
            },
            Object {
              "id": "cantabria",
              "provincias": Array [
                Object {
                  "id": 14,
                  "value": "Cantabria",
                },
              ],
              "value": "Cantabria",
            },
            Object {
              "id": "castilla_la_mancha",
              "provincias": Array [
                Object {
                  "id": 15,
                  "value": "Albacete",
                },
                Object {
                  "id": 16,
                  "value": "Ciudad Real",
                },
                Object {
                  "id": 17,
                  "value": "Cuenca",
                },
                Object {
                  "id": 18,
                  "value": "Guadalajara",
                },
                Object {
                  "id": 19,
                  "value": "Toledo",
                },
              ],
              "value": "Castilla la Mancha",
            },
            Object {
              "id": "castilla_y_leon",
              "provincias": Array [
                Object {
                  "id": 20,
                  "value": "Ávila",
                },
                Object {
                  "id": 21,
                  "value": "Burgos",
                },
                Object {
                  "id": 22,
                  "value": "Leon",
                },
                Object {
                  "id": 23,
                  "value": "Palencia",
                },
                Object {
                  "id": 24,
                  "value": "Salamanca",
                },
                Object {
                  "id": 25,
                  "value": "Segovia",
                },
                Object {
                  "id": 26,
                  "value": "Soria",
                },
                Object {
                  "id": 27,
                  "value": "Valladolid",
                },
                Object {
                  "id": 28,
                  "value": "Zamora",
                },
              ],
              "value": "Castilla y Leon",
            },
            Object {
              "id": "cataluna",
              "provincias": Array [
                Object {
                  "id": 29,
                  "value": "Barcelona",
                },
                Object {
                  "id": 30,
                  "value": "Gerona",
                },
                Object {
                  "id": 31,
                  "value": "Lerida",
                },
                Object {
                  "id": 32,
                  "value": "Tarragona",
                },
              ],
              "value": "Cataluña",
            },
            Object {
              "id": "ceuta",
              "provincias": Array [
                Object {
                  "id": 33,
                  "value": "Ceuta",
                },
              ],
              "value": "Ceuta",
            },
            Object {
              "id": "valencia",
              "provincias": Array [
                Object {
                  "id": 34,
                  "value": "Alicante",
                },
                Object {
                  "id": 35,
                  "value": "Castellón",
                },
                Object {
                  "id": 36,
                  "value": "Valencia",
                },
              ],
              "value": "Comunidad Valenciana",
            },
            Object {
              "id": "extremadura",
              "provincias": Array [
                Object {
                  "id": 37,
                  "value": "Badajoz",
                },
                Object {
                  "id": 38,
                  "value": "Cáceres",
                },
              ],
              "value": "Extremadura",
            },
            Object {
              "id": "baleares",
              "provincias": Array [
                Object {
                  "id": 39,
                  "value": "Islas Baleares",
                },
              ],
              "value": "Islas Baleares",
            },
            Object {
              "id": "galicia",
              "provincias": Array [
                Object {
                  "id": 40,
                  "value": "La Coruña",
                },
                Object {
                  "id": 41,
                  "value": "Lugo",
                },
                Object {
                  "id": 42,
                  "value": "Orense",
                },
                Object {
                  "id": 43,
                  "value": "Pontevedra",
                },
              ],
              "value": "Galicia",
            },
            Object {
              "id": "madrid",
              "provincias": Array [
                Object {
                  "id": 44,
                  "value": "Comunidad de Madrid",
                },
              ],
              "value": "Madrid",
            },
            Object {
              "id": "melilla",
              "provincias": Array [
                Object {
                  "id": 45,
                  "value": "Melilla",
                },
              ],
              "value": "Melilla",
            },
            Object {
              "id": "murcia",
              "provincias": Array [
                Object {
                  "id": 46,
                  "value": "Región de Murcia",
                },
              ],
              "value": "Murcia",
            },
            Object {
              "id": "navarra",
              "provincias": Array [
                Object {
                  "id": 47,
                  "value": "Comunidad foral de Navarra",
                },
              ],
              "value": "Navarra",
            },
            Object {
              "id": "pais_vasco",
              "provincias": Array [
                Object {
                  "id": 48,
                  "value": "Álava",
                },
                Object {
                  "id": 49,
                  "value": "Gipúzcoa",
                },
                Object {
                  "id": 50,
                  "value": "Vizcaya",
                },
              ],
              "value": "País Vasco",
            },
            Object {
              "id": "asturias",
              "provincias": Array [
                Object {
                  "id": 51,
                  "value": "Principado de Asturias",
                },
              ],
              "value": "Principado de asturias",
            },
          ]
        }
      />
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "black",
            "borderColor": "#d6d7da",
            "borderWidth": 0.5,
            "flexDirection": "row",
            "opacity": 0.5,
            "padding": 15,
            "width": "100%",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "width": "95%",
            }
          }
        >
          Provincia
        </Text>
        <Icon
          color="white"
          name="keyboard-arrow-down"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={20}
          type="MaterialIcons"
          underlayColor="white"
        />
      </TouchableOpacity>
      <SecondaryMenuModal
        handler={[Function]}
        hidePetTypeModal={[Function]}
        isVisible={false}
        items={Array []}
      />
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "grey",
            "borderColor": "#d6d7da",
            "borderWidth": 0.5,
            "flexDirection": "row",
            "opacity": 0.8,
            "padding": 15,
          }
        }
      >
        <Icon
          color="white"
          name="search"
          raised={false}
          reverse={false}
          reverseColor="white"
          size={30}
          style={
            Object {
              "marginLeft": "35%",
            }
          }
          type="evilIcons"
          underlayColor="white"
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "color": "white",
              "fontSize": 16,
            }
          }
        >
          Buscar
        </Text>
      </TouchableOpacity>
    </View>
  </ReactNativeModal>
</View>
`;
